pkg/pillar/cmd/volumemgr/handlevolume.go

```diff
@@ -289,40 +289,46 @@ func maybeDeleteVolume(ctx *volumemgrContext, status *types.VolumeStatus) {
 		log.Functionf("maybeDeleteVolume for %v Done", status.Key())
 		return
 	}
+	var readyToUnPublish bool
 	if status.SubState == types.VolumeSubStateCreated {
+		// we are not interested in result
+		_ = popVolumeWorkResult(ctx, status.Key())
+		status.SubState = types.VolumeSubStateDeleting
+		publishVolumeStatus(ctx, status)
 		// Asynch destruction; make sure we have a request for the work
 		AddWorkDestroy(ctx, status)
+	} else if status.SubState == types.VolumeSubStateDeleting {
 		vr := popVolumeWorkResult(ctx, status.Key())
 		if vr != nil {
 			log.Functionf("VolumeWorkResult(%s) location %s, created %t",
 				status.Key(), vr.FileLocation, vr.VolumeCreated)
-			if vr.VolumeCreated {
-				status.SubState = types.VolumeSubStateCreated
-				status.CreateTime = vr.CreateTime
+			if !vr.VolumeCreated {
+				readyToUnPublish = true
 			} else {
-				status.SubState = types.VolumeSubStateInitial
-			}
-			status.FileLocation = vr.FileLocation
-			if vr.Error != nil {
-				status.SetErrorWithSource(vr.Error.Error(),
```

Comment:
Was there a reason to remove the Source set and check in this logic?

Reply:
It looks like noop, we use it just ones to publish the status with that error and unpublish it. And we do not really resolve the error here, we just ignore it.